htdocs/expedition/card.php

```diff
@@ -1212,11 +1212,16 @@
 
 									print '<!-- Show details of lot -->';
 									print '<input name="batchl'.$indiceAsked.'_'.$subj.'" type="hidden" value="'.$dbatch->id.'">';
-									//print $langs->trans("DetailBatchFormat", $dbatch->batch, dol_print_date($dbatch->eatby,"day"), dol_print_date($dbatch->sellby,"day"), $dbatch->qty);
 									$productlotObject->fetch(0, $line->fk_product, $dbatch->batch);
-									print $langs->trans("Batch").': '.$productlotObject->getNomUrl(1);
-									print ' ('.$dbatch->qty.')';
-									//print $langs->trans("DetailBatchFormat", 'ee'.$dbatch->batch, dol_print_date($dbatch->eatby,"day"), dol_print_date($dbatch->sellby,"day"), $dbatch->qty);
+									if (!empty($productlotObject->batch)) 
+									{
+										print $langs->trans("Batch").': '.$productlotObject->getNomUrl(1);
+										print ' ('.$dbatch->qty.')';
+									}
+									else
+									{
+										print $langs->trans("DetailBatchFormat", $dbatch->batch, dol_print_date($dbatch->eatby,"day"), dol_print_date($dbatch->sellby,"day"), $dbatch->qty);
```

Comment:
This was disabled because content is not reliable.
The eatby and sellby is reliable only when stored into the table llx_product_lot (that is not erased with time, even if stock become 0, and that depend only on couple product/lot).
To have data from "old record" shown, the direction i would like user/integrator take is to clean/update/migrate their data into table llx_product_lot, that's why i don't think this PR is a good idea.
If the PR is only to show "batch" number, a simple 

> print $dbatch->batch.' ('.$dbatch->qty.')';

should be enough

Reply:
Yes batch number is enough. We can remove this patch in a later dolibarr version. Smaller companies don't need or will spend time to clean/update/migrate, they will consume the old batches.